id3: Remove various mutagen hacks which no longer work with newer mutagen

lazka opened the pull request:
Three parts:

(1) The monkey patching of EncodedTextSpec to ignore encoding errors
when using latin1 as encoding.

This gets replaced with a new id3text function which makes sure
that any value passed to mutagen can be encoded with the selected
encoding

(2) MultiSpec monkey patch to not null terminate a list of strings.

While this is valid according to the spec mutagen decided to terminate
all strings (not only the text lists handled here). If this is a
problem it should be discussed upstream (mutagen).

(3) compatid3 which implemented id3v2.3 support and added some
additional frames.

This gets replaced with the id3v2.3 support implemented upstream.
The additional frames still get passed to mutagen and, to match
the previous implementation, update_to_v23() gets wrapped to
allow some v2.4 only frames.

Freso commented on picard/formats/id3.py:
```diff
@@ -90,6 +57,16 @@ def patched_MultiSpec_write(self, frame, value):
 
+def id3text(text, encoding):
+    """Returns a string which only contains code points which can
+    be encododed with passed id3 encoding.
```
Typo.

phw commented:
Thanks a lot for this. I am glad this code got cleaned up, and looks good to me.

Not too sure about the implications of the string termination, but as you said if this has any issues it should be discussed upstream. I'd say we merge it this way for now, unless somebody is aware of any specific issues with this.

Just for reference: [PICARD-833](http://tickets.musicbrainz.org/browse/PICARD-833)

phw commented:
One question: Does this raise the minimum supported mutagen version? If so, we should document this.

Apart from this once the conflicts are resolved and the typo is fixed I think this is good to merge.

lazka commented:
(sorry, was without Internet for a few days)

> One question: Does this raise the minimum supported mutagen version? If so, we should document this.

The newly added code depends on 1.22+

lazka commented:
I'll make a 1.34.1 release including this: https://github.com/quodlibet/mutagen/commit/681a37ff1977996d3d65de7208d7f734840d1c0c

lazka commented:
New release is out: https://mutagen.readthedocs.io/en/latest/changelog.html

Freso commented:
Picard had stopped running for me. Running it patched with this PR makes it work again.

zas commented:
> The newly added code depends on 1.22+

So it raises the minimum version required, not much an issue, but https://github.com/metabrainz/picard/blob/master/INSTALL.txt#L15 should be modified accordingly.

lazka commented:
> So it raises the minimum version required, not much an issue, but https://github.com/metabrainz/picard/blob/master/INSTALL.txt#L15 should be modified accordingly.

Updated the INSTALL and added a note in the commit message

lazka commented:
Thanks!
